Atangster/282 add navigation E2E test

atangster opened the pull request:
## Pull Request Details

Link to associated issue: https://github.com/anusii/moviestar/issues/282

# Checklist
Complete the check-list below to ensure your branch is ready for PR.

Flutter Style Guide: https://survivor.togaware.com/gnulinux/flutter-style.html

- [x] Screenshots included in linked issue
- [x] Changes adhere to the team style and coding guideline
- [x] No confidential information
- [x] No duplicated content
- [x] No lint check errors related to your changes (make prep or flutter analyze lib)
- [x] Tested on at least one device
  - [x] Windows
  - [x] Linux
- [x] Added 2 reviewers (or 1 for private repositories then they add another)

atangster committed:
Add E2E test for navigation and state persistence

Implements comprehensive end-to-end testing for navigation flows
and state persistence across the MovieStar application.

## Test Coverage (9 tests)

### Tab Navigation Tests
- Navigate through all 8 main tabs sequentially
- Verify tab selection persists after modal navigation
- Verify all tab icons and titles render correctly

### Deep Navigation Tests
- Navigate from Home to movie details and back
- Test search functionality navigation flow
- Navigate to custom list details from My Lists

### View Mode Persistence Tests
- Verify view mode persists across tab navigation
- Test view mode restoration after app restart
- Cycle through all view modes and verify persistence

## Features

- POD authentication via CredentialInjector
- Safe Hive adapter registration (prevents duplicates)
- Icon-based navigation (compatible with SolidScaffold)
- Proper cleanup in tearDownAll
- Auto-regeneration of expired credentials

## Usage

```bash
# Run the test
flutter test integration_test/workflows/navigation_state_test.dart -d windows
```

Implements #282

atangster committed:
Add E2E integration test for navigation and state persistence

atangster committed:
Improve navigation test: add actual navigation and fix comment formatting

atangster committed:
improve comment format

atangster committed:
Merge branch 'atangster/302_fix_solidpod_api_migration' into atangster/282_nav_e2e_test_from_302

anushkavidanage reviewed (approved):
All worked well. Thanks @atangster

atangster committed:
Remove legacy authentication code from integration tests

atangster committed:
run dart format

atangster committed:
fix flutter analyze

atangster committed:
Reorganize and expand integration test documentation

- Move docs from docs/ to integration_test/docs/
- Rename files: TESTING_GUIDE.md -> testing-guide.md, INTEGRATION_TEST_SETUP.md -> setup-guide.md
- Add new comprehensive documentation:
  * README.md - Documentation index and navigation
  * authentication.md - OAuth, DPoP, RSA concepts with diagrams
  * architecture.md - Component diagrams and data flow
  * json-files.md - Complete JSON file structure reference
  * adapting.md - Guide for reusing in other POD apps
- Add Mermaid sequence/component diagrams for visual clarity
- Update all cross-references to new locations
- Add prominent links to conceptual docs in existing guides
- Reference external resources (RFCs, Solid specs) instead of duplicating
- Professional technical writing style without emojis

Documentation now answers:
- Why OAuth/DPoP instead of normal Flutter testing?
- What are the JSON files and when do they regenerate?
- How to adapt for other Solid POD applications?
- Token lifecycle and expiration handling

atangster committed:
Fix markdown linting issues and add link checker ignore

- Fix README.md list style (use + instead of -)
- Fix README.md line length (wrap long lines)
- Add sii.anu.edu.au to .lycheeignore (frequent CI timeouts)
- Address markdownlint MD004, MD013, MD032 violations

atangster committed:
Add TASK.md for markdown linting and file splitting work

Document the plan to:
- Fix markdown linting violations in all docs
- Split large files (testing-guide, adapting, architecture)
- Create 4 new focused documentation files
- Ensure all files are under 300 lines and lint-compliant

This is preparatory work - actual implementation to follow.

atangster committed:
Fix markdown linting and split large documentation files

atangster committed:
Reorganize documentation into subfolders (concepts, guides, reference)

atangster committed:
Remove TASK.md (work completed)

atangster committed:
Update README.md links to reflect documentation subfolder structure

gjwgit committed:
Merge pull request #309 from anusii/atangster/308_update_docs

Atangster/308 Reorganise and expand integration test documentation

gjwgit committed:
Merge pull request #307 from anusii/atangster/306_remove_legacy

Atangster/306 Remove legacy authentication code from integration tests

gjwgit committed:
Merge branch 'dev' into atangster/282_nav_e2e_test_from_302
